code/modules/antagonists/vampire/vampire_powers/dantalion_powers.dm

```diff
@@ -0,0 +1,192 @@
+		to_chat(target, "<span class='warning'>You feel the tendrils of evil invade your mind.</span>")
+		if(do_mob(user, target, 5 SECONDS))
+			if(can_enthrall(user, target) && vampire.subclass.thrall_cap > user.mind.som.serv.len)
+				handle_enthrall(user, target)
+				var/datum/spell_handler/vampire/V = custom_handler
+				var/blood_cost = V.calculate_blood_cost(vampire)
+				vampire.bloodusable -= blood_cost //we take the blood after enthralling, not before
+			else
+				revert_cast(user)
+				to_chat(user, "<span class='warning'>You or your target either moved or you dont have enough usable blood.</span>")
+
+/obj/effect/proc_holder/spell/vampire/enthrall/proc/can_enthrall(mob/living/user, mob/living/carbon/C)
+	if(!C)
+		log_runtime(EXCEPTION("something bad happened on enthralling a mob, attacker is [user] [user.key] \ref[user]"), user)
+		return FALSE
+	if(!ishuman(C))
+		to_chat(user, "<span class='warning'>You can only enthrall sentient humanoids!</span>")
+		return FALSE
+	if(!C.mind)
+		to_chat(user, "<span class='warning'>[C.name]'s mind is not there for you to enthrall.</span>")
+		return FALSE
+
+	var/enthrall_safe = FALSE
+	for(var/obj/item/implant/mindshield/L in C)
+		if(L && L.implanted)
+			enthrall_safe = TRUE
+			break
+	var/datum/antagonist/vampire/V = user.mind.has_antag_datum(/datum/antagonist/vampire)
+	if(V.subclass.thrall_cap <= user.mind.som.serv.len)
+		to_chat(user, "<span class='warning'>You don't have enough power to enthrall anymore people!</span>")
+		return FALSE
+	if(enthrall_safe || C.mind.has_antag_datum(/datum/antagonist/vampire) || C.mind.has_antag_datum(/datum/antagonist/mindslave))
+		C.visible_message("<span class='warning'>[C] seems to resist the takeover!</span>", "<span class='notice'>You feel a familiar sensation in your skull that quickly dissipates.</span>")
+		return FALSE
+	if(C.mind.isholy)
+		C.visible_message("<span class='warning'>[C] seems to resist the takeover!</span>", "<span class='notice'>Your faith in [SSticker.Bible_deity_name] has kept your mind clear of all evil.</span>")
+		return FALSE
+	return TRUE
+
+/obj/effect/proc_holder/spell/vampire/enthrall/proc/handle_enthrall(mob/living/user, mob/living/carbon/human/H)
+	if(!istype(H))
+		return FALSE
+
+	var/greet_text = "You have been Enthralled by [user.real_name]. Follow [user.p_their()] every command."
+	H.mind.add_antag_datum(new /datum/antagonist/mindslave/thrall(user.mind, greet_text))
+	H.Stun(2)
+	add_attack_logs(user, H, "Vampire-thralled")
+
+/obj/effect/proc_holder/spell/vampire/thrall_commune
+	name = "Commune"
+	desc = ":^ Thrall gang lmao"
+	gain_desc = "You have gained the ability to commune with your thralls."
+
+/datum/spell_targeting/select_vampire_master/choose_targets(mob/user, obj/effect/proc_holder/spell/spell, params, atom/clicked_atom) // finds the master of the thralls, returns the user if the vampire is using it
```

Comment:
```suggestion
/datum/spell_targeting/select_vampire_network/choose_targets(mob/user, obj/effect/proc_holder/spell/spell, params, atom/clicked_atom) // Returns the vampire and their thralls. If user is a thrall then it will look up their master's network
```
Something the like would be a bit more descriptive.
You can change it to something else btw. The old version is just not that descriptive to what it actually does